### ISSUE TITLE
Table created by stock cassandra-stress 2.1.8 is not marked with compact storage

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylla#402]
Client:
```
apache-cassandra-2.1.8/tools/bin/cassandra-stress write n=1 -mode native cql3
```

On Cassandra 2.1.8:

```
$ bin/cqlsh -e "describe COLUMNFAMILY  keyspace1.standard1"

CREATE TABLE keyspace1.standard1 (
    key blob PRIMARY KEY,
    "C0" blob,
    "C1" blob,
    "C2" blob,
    "C3" blob,
    "C4" blob
) WITH COMPACT STORAGE
    AND caching = '{"keys":"ALL", "rows_per_partition":"NONE"}'
    AND comment = ''
    AND compaction = {'class': 'org.apache.cassandra.db.compaction.SizeTieredCompactionStrategy'}
    AND compression = {}
    AND dclocal_read_repair_chance = 0.1
    AND default_time_to_live = 0
    AND gc_grace_seconds = 864000
    AND max_index_interval = 2048
    AND memtable_flush_period_in_ms = 0
    AND min_index_interval = 128
    AND read_repair_chance = 0.0
    AND speculative_retry = 'NONE';
```

On Scylla 0.9:

```
$ bin/cqlsh -e "describe COLUMNFAMILY  keyspace1.standard1"

CREATE TABLE keyspace1.standard1 (
    key blob PRIMARY KEY,
    "C0" blob,
    "C1" blob,
    "C2" blob,
    "C3" blob,
    "C4" blob
) WITH bloom_filter_fp_chance = 0.01
    AND caching = '{"keys":"ALL","rows_per_partition":"ALL"}'
    AND comment = ''
    AND compaction = {'class': 'SizeTieredCompactionStrategy'}
    AND compression = {}
    AND dclocal_read_repair_chance = 0.1
    AND default_time_to_live = 0
    AND gc_grace_seconds = 864000
    AND max_index_interval = 2048
    AND memtable_flush_period_in_ms = 0
    AND min_index_interval = 128
    AND read_repair_chance = 0.0
    AND speculative_retry = '99.0PERCENTILE';
```

**Comment by penberg** [comment]
@tgrabiec @raphaelsc The CQL front end should be OK.

I am not really an expert on this area of the code but ``schema_builder::build()`` has this check:

```
s._raw._is_dense = (cp == compact_storage::yes) && (s.clustering_key_size() > 0);
```

AFAICT, ``standard1`` has no clustering key components so we ``_is_dense`` will be set to false and thus the sstable code will never know about it.

**Comment by nyh** [comment]
On Thu, Sep 24, 2015 at 4:23 PM, Pekka Enberg <notifications@github.com>
wrote:

> @tgrabiec <https://github.com/tgrabiec> @raphaelsc
> <https://github.com/raphaelsc> The CQL front end should be OK.
>
> I am not really an expert on this area of the code but
> schema_builder::build() has this check:
>
> s._raw._is_dense = (cp == compact_storage::yes) && (s.clustering_key_size() > 0);
>
> AFAICT, standard1 has no clustering key components so we _is_dense will
> be set to false and thus the sstable code will never know about it.
>

But such logic would be wrong -  if "with compact storage" is not
specified, composite key names are always used, regardless of whether or
not there are clustering columns (s.clustering_key_size() > 0).

-- 
Nadav Har'El
nyh@cloudius-systems.com

**Comment by penberg** [comment]
According to @tgrabiec, the problem does not happen with CQL at all, but with Thrift. And indeed, looking at ``thrift_handler::system_add_keyspace()``, we don't even use ``schema_builder`` so we don't have the chance to set compact storage option.

@slivne Is this really a 0.9 issue? I thought we're not doing Thrift work...

**Comment by tgrabiec** [comment]
Should be fixed by 41843b32c5d7e7a7d1c6e962dc44507d5249fdeb
